enhancements/cluster-logging/cluster-logging-v2-apis.md

```diff
@@ -0,0 +1,404 @@
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+
+# v2 API for Log Forwarding and Log Storage
+## Summary
+
+Logging for Red Hat OpenShift has evolved since its initial release in OpenShift 3.x from an on-cluster, highly opinionated offering to a more flexible log forwarding solution that supports multiple internal (e.g LokiStack, Elasticsearch) and externally managed log storage.  Given the original components (e.g. Elasticsearch, Fluentd) have been deprecated for various reasons, this enhancement introduces the next version of the APIs in order to formally drop support for those features as well as to generally provide an API to reflect the future direction of log storage and forwarding.
+
+## Motivation
+
+### User Stories
+
+The next version of the APIs should continue to support the primary objectives of the project which are: 
+
+* Collect logs from various sources and services running on a cluster
+* Normalize the logs to common format to include workload metadata (i.e. labels, namespace, name)
+* Forward logs to storage of an administrator's choosing (e.g. LokiStack)
+* Provide a Red Hat managed log storage solution
+* Provide an interface to allow users to review logs from a Red Hat managed storage solution
+
+The following user stories describe deployment scenarios to support these objectives:
+
+* As an administrator, I want to deploy a complete operator managed logging solution that includes collection, storage, and visualization so I can evaluate log records while on the cluster
+* As an administrator, I want to deploy an operator managed log collector only so that I can forward logs to an existing storage solution
+* As an administrator, I want to deploy an operator managed instance of LokiStack and visualization
+
+The administrator role is any user who has permissions to deploy the operator and the cluster-wide resources required to deploy the logging components.
+
+### Goals
+
+* Drop support for the **ClusterLogging** custom resource
+* Drop support for **ElasticSearch**, **Kibana** custom resources and the **elasticsearch-operator**
+* Drop support for Fluentd collector implementations, Red Hat managed Elastic stack (e.g. Elasticsearch, Kibana)
+* Drop support in the **cluster-logging-operator** for **log-view-plugin** management 
+* Support log forwarder API with no dependency upon reserved words (e.g. default)
+* Support an API to spec a Red Hat managed LokiStack with the logging tenancy model
+* Support an API to allow flexible deployment of the logging components: collector/forwarder, storage, visualization
+* Continue to allow deployment of a log forwarder to the output sinks of the administrators choosing
+
+
+### Non-Goals
+
+* "One click" deployment of a full logging stack as provided by **ClusterLogging** v1
+* Complete backwards compatibility to **ClusterLogForwarder** v1
+* Automated migration path from v1 to v2
```

Comment:
Are v1 and v2 CRDs supported at the same time or is there a hard cutover required?

Reply:
IMO, ideally a hard cut over but there are open questions about what it means for users not heeding the deprecation warnings to come to the next version of logging with Elasticsearch and fluentd.  Mostly, the v2 api for  vector deployments maps in a straight forward fashion from v1

Reply:
@jewzaam further discussion.  Updated the proposal to land on a path that:
* initially migrates the resource
* drops support for previous version in subsequent release stream

Reply:
Updated again:
* drops CL objects
* allows v1 and v2 for vector CLF deployments
* migrates v1 to v2
* terminates new feature development for v1